src/TagField.php

```diff
@@ -402,6 +402,14 @@ protected function getListValues($values): array
             return $values->column($this->getTitleField());
         }
 
+        if ($values instanceof DataObject && $values->exists()) {
+            return [$values->getField($this->getTitleField())];
```

Comment:
`getField()` will skip any `getTitle()` method that might be intended to override the db value.
```suggestion
            return [$values->{$this->getTitleField()}];
```

Reply:
Thanks for the review. Changed as suggested

Reply:
Awesome, thanks for that. Can you please address the ID fallback as well?

Reply:
@GuySartorelli done

Reply:
Thanks!